### PR TITLE
MAKAIRA-3566 Multi-Tenant functionality

### PR DESCRIPTION
**Pull request by nfqphihungnguyen**

### CODE REVIEW

File: src/Repository/AppInfoRepository.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+//            ->andWhere('a.exampleField = :val')
+//            ->setParameter('val', $value)
+//            ->orderBy('a.id', 'ASC')
+//            ->setMaxResults(10)
+//            ->getQuery()
+//            ->getResult()
+//        ;
+//    }
+
+   public function findOneBySome(string $domain, string $instance): ?AppInfo
```
Comment (EricTietz) [inline src/Repository/AppInfoRepository.php]: Better to name it to something specific, like `findOneByDomainAndInstance`, i believe :)

File: src/Entity/AppInfo.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+#[ORM\Entity(repositoryClass: AppInfoRepository::class)]
+class AppInfo
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column]
+    private ?int $id = null;
+
+    #[ORM\Column(length: 255)]
```
Comment (EricTietz) [inline src/Entity/AppInfo.php]: Shouldn't those columns be nullable? If so, then you would need to express it specifically, as the default of "nullable" = false , Isn't it?

File: src/Security/SignedRequestAuthenticator.php [hunk 3]
```diff
@@ -14,17 +14,19 @@
...
 use Twig\Error\SyntaxError;
+use App\Entity\AppInfo;
+use Doctrine\Persistence\ManagerRegistry;
 
 class SignedRequestAuthenticator extends AbstractAuthenticator
 {
 
     private Environment $twig;
-    private string $clientSecret;
+    private ManagerRegistry $doctrine;
```
Comment (EricTietz) [inline src/Security/SignedRequestAuthenticator.php]: You don't need to get the Managerregistry here, just inject the repository itself ^^

File: src/Security/SignedRequestAuthenticator.php [hunk 4]
```diff
@@ -57,10 +59,12 @@ public function authenticate(Request $request): Passport
             throw new AuthenticationException();
         }
 
+        $appInfo = $this->doctrine->getRepository(AppInfo::class)->findOneBySome($domain, $instance);
```
Comment (EricTietz) [inline src/Security/SignedRequestAuthenticator.php]: when injecting the repo itself in the constructor, you can access it here directly via $this->appInfoRepository->...
Not going via the "getRepository"-Method of the Managerregistry is actually a concern called "decoupling"

File: src/Service/CommunicationService.php [hunk 5]
```diff
@@ -3,18 +3,21 @@
...
 class CommunicationService
 {
     private string $nonce;
+    private ManagerRegistry $doctrine;
 
     /**
      * @throws Exception
      */
-    public function __construct(private readonly string $clientSecret, private readonly HttpClient $httpClient, private readonly string $url, private readonly string $instance)
+    public function __construct(ManagerRegistry $doctrine)
```
Comment (EricTietz) [inline src/Service/CommunicationService.php]: some as above, inject the Repository directly

File: src/Entity/AppInfo.php (hunk 2 above)
Comment (nfqphihungnguyen) [reply src/Entity/AppInfo.php]: The migrations of those columns are NOT NULL by default, but ok I will make it looks more properly :)

Comment (EricTietz) [reply src/Entity/AppInfo.php]: Ah, ok...nevermind my comment then, not so important then, my mistake

File: src/Entity/AppInfo.php [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+#[ORM\Entity(repositoryClass: AppInfoRepository::class)]
+class AppInfo
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column(nullable: false)]
+    private ?int $id = null;
+
+    #[ORM\Column(length: 255, nullable: false)]
+    private ?string $makairaDomain = null;
```
Comment (EricTietz) [inline src/Entity/AppInfo.php]: I still don't get, why this Column is not nullable, but you set the dafult Value to null? Isn't that contradictory? If you want an empty default value, wouldn't an empty string be better?

File: src/Repository/AppInfoRepository.php (hunk 1 above)
Comment (nfqphihungnguyen) [reply src/Repository/AppInfoRepository.php]: Updated

File: src/Security/SignedRequestAuthenticator.php (hunk 3 above)
Comment (nfqphihungnguyen) [reply src/Security/SignedRequestAuthenticator.php]: Updated

File: src/Security/SignedRequestAuthenticator.php (hunk 4 above)
Comment (nfqphihungnguyen) [reply src/Security/SignedRequestAuthenticator.php]: Updated

File: src/Service/CommunicationService.php (hunk 5 above)
Comment (nfqphihungnguyen) [reply src/Service/CommunicationService.php]: Updated

**Pull request merged**